src/node_crypto.cc

```diff
@@ -3871,24 +3873,26 @@ SignBase::Error Sign::SignFinal(const char* key_pem,
     goto exit;
 
 #ifdef NODE_FIPS_MODE
-  /* Validate DSA2 parameters from FIPS 186-4 */
-  if (EVP_PKEY_DSA == pkey->type) {
-    size_t L = BN_num_bits(pkey->pkey.dsa->p);
-    size_t N = BN_num_bits(pkey->pkey.dsa->q);
-    bool result = false;
-
-    if (L == 1024 && N == 160)
-      result = true;
-    else if (L == 2048 && N == 224)
-      result = true;
-    else if (L == 2048 && N == 256)
-      result = true;
-    else if (L == 3072 && N == 256)
-      result = true;
-
-    if (!result) {
-      fatal = true;
-      goto exit;
+  if (FIPS_mode()) {
+    /* Validate DSA2 parameters from FIPS 186-4 */
+    if (EVP_PKEY_DSA == pkey->type) {
```

Comment:
Just for the sake of saving `git blame` data. Could we merge it into one `if` with `&&`?